agent/engine/task_manager_test.go

```diff
@@ -2227,3 +2231,157 @@ func TestTaskWaitForHostResources(t *testing.T) {
+		Task: &apitask.Task{
+			ResourcesMapUnsafe:  make(map[string][]taskresource.TaskResource),
+			DesiredStatusUnsafe: apitaskstatus.TaskRunning,
+			Volumes: []apitask.TaskVolume{
+				{
+					Name: taskresourcevolume.TestVolumeName,
+					Type: apiresource.EBSTaskAttach,
+					Volume: &taskresourcevolume.EBSTaskVolumeConfig{
+						VolumeId:             taskresourcevolume.TestVolumeId,
+						VolumeName:           taskresourcevolume.TestVolumeId,
+						VolumeSizeGib:        taskresourcevolume.TestVolumeSizeGib,
+						SourceVolumeHostPath: taskresourcevolume.TestSourceVolumeHostPath,
+						DeviceName:           taskresourcevolume.TestDeviceName,
+						FileSystem:           taskresourcevolume.TestFileSystem,
+					},
+				},
+			},
+		},
+		ctx:                      ctx,
+		resourceStateChangeEvent: make(chan resourceStateChange),
+	}
+
+	timeoutctx, timeoutcancel := context.WithTimeout(ctx, 50*time.Millisecond)
+	defer timeoutcancel()
+
+	mockCsiClient := mock_csiclient.NewMockCSIClient(mockCtrl)
+	mockCsiClient.EXPECT().NodeUnstageVolume(gomock.Any(), "vol-12345", "/mnt/ecs/ebs/taskarn_vol-12345").Do(func(ctx context.Context, volID, hostPath string) {
+		time.Sleep(5 * time.Millisecond)
+	}).Return(nil).MinTimes(1)
+
+	errors := mtask.UnstageVolumes(mockCsiClient)
+
+	require.NoError(t, timeoutctx.Err())
+	assert.Len(t, errors, 0)
+}
+
+func TestUnstageVolumesTimeoutError(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+	mtask := &managedTask{
+		Task: &apitask.Task{
+			ResourcesMapUnsafe:  make(map[string][]taskresource.TaskResource),
+			DesiredStatusUnsafe: apitaskstatus.TaskRunning,
+			Volumes: []apitask.TaskVolume{
+				{
+					Name: taskresourcevolume.TestVolumeName,
+					Type: apiresource.EBSTaskAttach,
+					Volume: &taskresourcevolume.EBSTaskVolumeConfig{
+						VolumeId:             taskresourcevolume.TestVolumeId,
+						VolumeName:           taskresourcevolume.TestVolumeId,
+						VolumeSizeGib:        taskresourcevolume.TestVolumeSizeGib,
+						SourceVolumeHostPath: taskresourcevolume.TestSourceVolumeHostPath,
+						DeviceName:           taskresourcevolume.TestDeviceName,
+						FileSystem:           taskresourcevolume.TestFileSystem,
+					},
+				},
+			},
+		},
+		ctx:                      ctx,
+		resourceStateChangeEvent: make(chan resourceStateChange),
+	}
+
+	timeoutctx, timeoutcancel := context.WithTimeout(ctx, 5*time.Millisecond)
+	defer timeoutcancel()
+
+	mockCsiClient := mock_csiclient.NewMockCSIClient(mockCtrl)
+	mockCsiClient.EXPECT().NodeUnstageVolume(gomock.Any(), "vol-12345", "/mnt/ecs/ebs/taskarn_vol-12345").Do(func(ctx context.Context, volID, hostPath string) {
+		time.Sleep(50 * time.Millisecond)
+	}).Return(fmt.Errorf("rpc error: code = DeadlineExceeded desc = context deadline exceeded")).MinTimes(1)
+
+	errors := mtask.UnstageVolumes(mockCsiClient)
+
+	require.Error(t, timeoutctx.Err())
```

Comment:
Where is `timeoutctx` used?

Reply:
oh! good catch

Reply:
actually nevermind, we're using it to check if we hit a context deadline exceeded error.

Reply:
I think it's unused as we only use it in an assertion.